### DISCUSSION TITLE
ddns-updater alternating between actual WAN IP and VPN side WAN IP

### DISCUSSION THREAD

**Discussion by pullasuti** [qdm12/ddns-updater#827] [Q&A]
I'm having an issue with ddns-updater picking up a second VPN address instead of my actual IP. The address is alternating between actual WAN IP and a Wireguard setup on wifi router (AsusWRT & Wireguard). This doesn't cause issues with a "backup" ddns script on the router to No-IP, but ddns-updater keeps switching the IPs every 20 minutes :confused: 

I have had the same setup since I started using ddns-updater, with the only change being that I added a second VPN tunnel on my router recently. Currently both tunnels are configured similarly, but somehow the WAN IP at the end of the second VPN connection is still picked by ddns-updater. 

Not worth an "issue" since this is a "temporary" VPN setup anyway (and probably my fault), with a better multisite setup hopefully up some day. Just curious if there is something I could try with ddns-updater, or just disable the troublesome VPN connection for now?

**Comment by qdm12** [comment]
Ddns-updater queries ip echo services such as ipinfo.io/ip to get the public ip address. I guess its network just goes through the vpn tunnel for some reason?

How are you running ddns-updater? As a standalone binary?

**Reply by pullasuti** [reply]
Yes, a standalone binary running from a Proxmox LXC (Alpine). Works fine when I have just one WG client set up on the router, but after enabling the second one, keeps swapping between the two IPs.

Well, a good reason to improve my VPN setup :)

**Reply by jablon-cube** [reply]
@pullasuti Have you found a fix?

DDNS-Updater has been running fine in a docker container for 2 years. Host is TrueNas.

Now I'm having the same issue described above where it alternates between WAN IP and Wireguard IP at every refresh. The constant notifications is forcing me to revert to my previous config.
Issue started after switching vpn protocols (from OpenVpn to Wireguard) on an Asus Merlin router. Traffic from the docker host is going through the WAN as it should. So are the other containers.

Any suggestion/help is much appreciated.

**Reply by pullasuti** [reply]
> @pullasuti Have you found a fix?
> 
> DDNS-Updater has been running fine in a docker container for 2 years. Host is TrueNas.
> 
> Now I'm having the same issue described above where it alternates between WAN IP and Wireguard IP at every refresh. The constant notifications is forcing me to revert to my previous config. Issue started after switching vpn protocols (from OpenVpn to Wireguard) on an Asus Merlin router. Traffic from the docker host is going through the WAN as it should. So are the other containers.
> 
> Any suggestion/help is much appreciated.

Sorry, not really. I finally moved my Wireguard off of AsusWRT/Merlin, but have yet to set up a permanent replacement (thank you TP-Link for making routers with the inability to assign actual routes! :smiling_face_with_tear:  ). 

I was thinking maybe Wireguard was passing DNS traffic over the VPN a little too eagerly due to a "misunderstanding" with the config (allowed IPs vs. all DNS queries going through WG), but can't verify this any longer (and this is also weird, because the address still kept flip-flopping back and forth).
